src/Searcher.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from "react";
+import * as _ from "underscore";
+import { LibraryItem } from "./components/LibraryItem";
+import { SearchResultItem } from "./components/SearchResultItem";
+import { LibraryContainer } from "./components/LibraryContainer";
+import { searchItemResursive, setItemStateRecursive, ItemData } from "./LibraryUtilities";
+import { SearchBar } from "./components/SearchBar";
+
+type displayMode = "structure" | "list";
+
+interface SearchModeChangedFunc {
+    (inSearchMode: boolean): void;
+}
+
+interface SearcherProps {
+    onSearchModeChanged: SearchModeChangedFunc;
+    libraryContainer: LibraryContainer;
+    sections: ItemData[];
+    categories: string[];
+}
+
+interface SearcherStates {}
+
+export class Searcher extends React.Component<SearcherProps, SearcherStates> {
```

Comment:
`Searcher` has no reason to be a `React.Component`, it should be a regular TypeScript class so unit test cases can be written for it with mocha/chai frameworks.